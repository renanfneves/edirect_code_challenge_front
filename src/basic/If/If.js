const If = ({ condition, children }) => (condition ? children : null);

export default If;
