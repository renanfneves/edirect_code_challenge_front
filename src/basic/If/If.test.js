import React from "react";
import ShallowRenderer from "react-test-renderer/shallow";
import If from "./If";

describe("If - Util", () => {
  it("Should render child", () => {
    // given
    const renderer = new ShallowRenderer();

    // when
    renderer.render(
      <If condition>
        <div id="condition">condition</div>
      </If>,
    );
    const result = renderer.getRenderOutput();

    // then
    expect(result).toMatchSnapshot();
  });

  it("should not render child", () => {
    // given
    const renderer = new ShallowRenderer();

    // when
    renderer.render(
      <If condition={false}>
        <div id="condition">condition</div>
      </If>,
    );
    const result = renderer.getRenderOutput();

    // then
    expect(result).toMatchSnapshot();
  });
});
