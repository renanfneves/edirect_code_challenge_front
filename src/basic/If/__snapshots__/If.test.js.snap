// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`If - Util Should render child 1`] = `
<div
  id="condition"
>
  condition
</div>
`;

exports[`If - Util should not render child 1`] = `null`;
